wfe2/wfe_test.go

```diff
@@ -2123,3 +2135,28 @@ func TestDeactivateRegistration(t *testing.T) {
 		  "status": 403
 		}`)
 }
+
+func TestNewOrder(t *testing.T) {
```

Comment:
I think this unit test is a good start but I'd like to see a couple of the `TestIssueCertificate` cases ported over to `NewOrder` if you could. `TestIssueCertificate`  has test cases for malformed POSTs, broken signatures on the CSR, and a CSR generated with the old OpenSSL. The latter two seem like they should be tested for. I'm on the fence about the WFE tests needing to duplicate the same "no post body, no jws, bad jws, etc etc" that many of them do now.